src/components/RecordTypesSelect/components/ChooseExistingRecord.js

```diff
@@ -0,0 +1,168 @@
+
+    if (selectedExistingRecordElem) {
+      setTop(selectedExistingRecordElem.offsetTop);
+      setHeight(selectedExistingRecordElem.getBoundingClientRect().height);
+    }
+  }, [existingRecord, compareElemWidth, hintElemWidth]);
+
+  const render = typeof top === 'number';
+
+  return (
+    <section
+      className={css.container}
+      data-test-choose-existing-record
+    >
+      <Row>
+        <Col
+          xs={4}
+          className={classNames(css.headingCell)}
+        >
+          <Headline
+            size="large"
+            margin="none"
+            tag="h3"
+          >
+            <FormattedMessage id="ui-data-import.incomingMarc" />
+          </Headline>
+        </Col>
+        <Col
+          xs
+          className={classNames(css.headingCell, css.borderLeft)}
+        >
+          <Headline
+            size="large"
+            margin="none"
+            tag="h3"
+          >
+            <FormattedMessage id="ui-data-import.existingRecords" />
+          </Headline>
+        </Col>
+      </Row>
+      <Row className={css.contentContainer}>
+        <div
+          className={css.selectedLine}
+          style={{
+            visibility: render ? null : 'hidden',
+            top,
+            height: `calc(${height}px + ${css.cellPadding})`,
+          }}
+        >
+          <div
+            ref={hintElemRef}
+            className={css.hint}
+          >
+            <Triangle
+              direction="left"
+              color="#616161"
+            />
+            <div className={css.hintMessage}>
+              <FormattedMessage id="ui-data-import.compareHint" />
+            </div>
+          </div>
+        </div>
+        <Col
+          xs={4}
+          className={classNames(css.cell)}
+          style={{
+            visibility: render ? null : 'hidden',
+            marginTop: top,
+            paddingRight: spaceForCompareElem,
+          }}
+        >
+          <RecordItem
+            item={incomingRecord}
+            className={css.incomingRecord}
+            style={{ height }}
+            onClick={() => setIncomingRecord(null)}
+          />
+          <div
+            ref={compareElemRef}
+            className={css.compare}
+            style={{
+              width: compareElemWidth,
+              height: `calc(${height}px + ${css.cellPadding} - ${css.selectedLineBorderWidth} - ${css.selectedLineBorderWidth})`,
+            }}
+          >
+            <Triangle />
+            <span className={css.compareMessage}>
+              <FormattedMessage id="ui-data-import.compare" />
+            </span>
+            <Triangle direction="left" />
+          </div>
+        </Col>
+        <Col
+          xs
+          className={classNames(css.cell, css.borderLeft)}
+          style={{
+            paddingLeft: spaceForCompareElem,
+            paddingRight: hintElemWidth + 9 + 4 + 8,
```

Comment:
This is kind of magic number, could you please add a comment or better move them to variables.